### PR TITLE
cleanup gulp tasks & npm scripts

### PR DESCRIPTION
**Pull request by FredKSchott**
- 'init' & 'typings' are going away in #411
- remove 'build-all', now just use 'test'

/cc @justinfagnani @rictic

### CODE REVIEW

**Comment by justinfagnani** [comment]
That AppVeyor error looks odd, I restarted the build.

File: package.json [hunk 1]
```diff
@@ -13,8 +13,8 @@
     "lint": "gulp lint",
     "init": "gulp init",
     "build": "gulp build",
-    "test": "gulp test",
-    "prepublish": "gulp build-all && gulp test"
+    "test": "gulp lint test",
+    "prepublish": "gulp build && npm run test"
```
Comment (justinfagnani) [inline package.json]: `npm test`?

**Comment by justinfagnani** [comment]
Huh... that AppVeyor build is still failing. I don't get the error though. Can you hold off on this?

**Comment by FredKSchott** [comment]
Yea, very strange. I'll hold off if you want to take a look at it tomorrow.

fwiw this was originally based off of #411, but I refactored it off of it and onto master so that it wouldn't block.

File: package.json (hunk 1 above)
Comment (rictic) [inline package.json]: +1, npm test has better output on error.

**Comment by FredKSchott** [comment]
Updated to base off of master including #411, hopefully the new merge conflict solves whatever was wrong on appveyor

**Comment by justinfagnani** [comment]
Still failing on appveyor/branch, but with what looks like a depcheck problem now. Very strange. Can you rebase rather than merge so that /branch and /pr are identical in terms of commits?

**Comment by FredKSchott** [comment]
Looks like that fixed it!

**Comment by justinfagnani** [comment]
So weird. There were two very different looking failure going on. Oh well...

**Pull request merged**